### PR TITLE
Add a timestamp parsing function

### PR DESCRIPTION
**Pull request by mprahl**
This will be useful for when we parse the `since` input for our scrapers.

### CODE REVIEW

File: purview/utils/general.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+def timestamp_to_datetime(timestamp):
+    """
+    Parses a timestamp and return a datetime object
+    :param timestamp: a string of a timestamp
+    :return: a datetime object
+    """
+    log.debug('Trying to parse the timestamp "{0}"'.format(timestamp))
+    combinations = (
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2})(?:\.\d+)?$', '%Y-%m-%d %H:%M:%S'),
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2})$', '%Y-%m-%d'),
```
Comment (yashvardhannanavati) [inline purview/utils/general.py]: Just one concern, I don't think it will match something like 2016-3-3

Comment (mprahl) [reply purview/utils/general.py]: Good point. I'll tweak the regex.

File: purview/utils/general.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    :param timestamp: a string of a timestamp
+    :return: a datetime object
+    """
+    log.debug('Trying to parse the timestamp "{0}"'.format(timestamp))
+    error_msg = 'The timestamp "{0}" is an invalid format'.format(timestamp)
+    combinations = (
+        (r'^(?P<datetime>\d{4}-\d{1,2}-\d{1,2} \d{1,2}:\d{1,2}:\d{1,2})(?:\.\d+)?$',
+         '%Y-%m-%d %H:%M:%S'),
+        (r'^(?P<datetime>\d{4}-\d{1,2}-\d{1,2})$', '%Y-%m-%d'),
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})(?:\.\d+)?(?:Z|[-+]00(?::00)?)?$',
```
Comment (yashvardhannanavati) [inline purview/utils/general.py]: Can we do the same here as well? Or am I missing something?

Comment (mprahl) [reply purview/utils/general.py]: Not there because I'm following the ISO-8601 spec there.

Comment (yashvardhannanavati) [reply purview/utils/general.py]: ah, sounds good!

File: tox.ini [hunk 3]
```diff
@@ -1,11 +1,15 @@
 [tox]
-envlist = flake8
+envlist = flake8,py27,p36
```
Comment (yashvardhannanavati) [inline tox.ini]: (absolutely optional) Space missing after comma? :P

Comment (mprahl) [reply tox.ini]: Sure, let me try it!

Comment (mprahl) [reply tox.ini]: Fixed

**Comment by yashvardhannanavati** [comment]
+1

**Pull request merged**